### PR TITLE
ddtrace/mocktracer: (*mockspan).Finish must be idempotent

### PR DESCRIPTION
**Pull request by phoenix2x**
Currently mockspan.Finish is not idempotent while span.Finish is. This prevents us from testing code which invokes Finish multiple times. Fix for #572

### CODE REVIEW

**Comment by gbbr** [comment]
You'll have to also update `(internal/version/version.go).Tag` to `v1.21.0` to get CI to pass.

**Comment by gbbr** [comment]
Note that I've updated your PR with latest base to get CI to pass. Please make sure you sync locally.

File: ddtrace/mocktracer/mockspan.go [hunk 1]
```diff
@@ -105,12 +105,17 @@ type mockspan struct {
 	parentID  uint64
 	context   *spanContext
 	tracer    *mocktracer
+
+	finished bool
```
Comment (gbbr) [inline ddtrace/mocktracer/mockspan.go]: Maybe this is better suited straight under `finishTime`

**Comment by phoenix2x** [comment]
> Please forgive my follow-up review. Can you please address these last few nits? Happy to merge after.

Wow, you guys are really strict:)

**Pull request merged**